channeldb/channel.go

```diff
@@ -401,20 +401,36 @@ var (
 	// will have.
 	ChanStatusRestored ChannelStatus = 1 << 3
 
-	// ChanStatusCoopBroadcasted indicates that a cooperative close for this
-	// channel has been broadcasted.
+	// ChanStatusCoopBroadcasted indicates that a cooperative close for
+	// this channel has been broadcasted. Older cooperatively closed
+	// channels will only have this status set. Newer ones will also have
+	// close initiator information stored using the local/remote initiator
+	// status. This status is set in conjunction with the initiator status
+	// so that we do not need to check multiple channel statues for
+	// cooperative closes.
 	ChanStatusCoopBroadcasted ChannelStatus = 1 << 4
+
+	// ChanStatusLocalCoopInitiated is set in addition to
+	// ChanStatusCoopBroadcasted when we initiate a cooperative close.
+	ChanStatusLocalCoopInitiated ChannelStatus = 1 << 5
```

Comment:
cool, I like this :) Looking at the chan statuses, we don't actually store whether we initiate a force close. This could be useful information also, like if we initiate, but the remote force close ends up in the chain. So we could make this a bit more general, and store it also in the cases where we/them force close?

Reply:
Updated to include force closes. Not quite the same flow for local vs remote force closes because we have some warning for local force where we can mark broadcast and set the initiator status. We only detect remote forces once they've been broadcast, and we don't mark broadcast (because we use that field for rebroadcasting on startup).